src/Launcher.vala

```diff
@@ -381,6 +386,11 @@ public class Dock.Launcher : Gtk.Box {
         return MOVE;
     }
 
+    /**
+     * Calculates to which side of us source (the launcher that's currently being reordered)
+     * should be moved. This will depend on the direction from which the mouse cursor entered
+     * and whether source is already next to us.
+     */
```

Comment:
Could maybe benefit from using Valadoc style comments?
```suggestion
    /**
     * Calculates which side of #this source should be moved to.
     * Depends on the direction from which the mouse cursor entered
     * and whether source is already next to #this.
     *
     * @param source the launcher that's currently being reordered
     * @param x pointer x position
     * @param y pointer y position
     */
```

Reply:
Uuuuh that's really nice